test/Microsoft.Data.Sqlite.Tests/SqliteConnectionTest.cs

```diff
@@ -311,6 +312,67 @@ public void CreateCommand_returns_command()
         }
 
         [Fact]
+        public void CreateCollation_works()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateCollation("MY_NOCASE", (s1, s2) => string.Compare(s1, s2, StringComparison.OrdinalIgnoreCase));
+
+                Assert.Equal(1L, connection.ExecuteScalar<long>("SELECT 'Νικοσ' = 'ΝΙΚΟΣ' COLLATE MY_NOCASE;"));
+            }
+        }
+
+        [Fact]
+        public void CreateCollation_with_null_comparer_works()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateCollation("MY_NOCASE", (s1, s2) => string.Compare(s1, s2, StringComparison.OrdinalIgnoreCase));
+                connection.CreateCollation("MY_NOCASE", null);
+
+                var ex = Assert.Throws<SqliteException>(
+                    () => connection.ExecuteScalar<long>("SELECT 'Νικοσ' = 'ΝΙΚΟΣ' COLLATE MY_NOCASE;"));
+
+                Assert.Equal(raw.SQLITE_ERROR, ex.SqliteErrorCode);
+                Assert.Contains("no such collation sequence", ex.Message);
```

Comment:
Removed assertion. Adding a complete ExecuteReader loop for analyzing the PRAGMA seems to be overly complex. Since sqlite version 3.16.0 there are table-valued functions for the PRAGMAs usable in SELECTs, but I don't think that we can rely on the use of such a new version.